xdsl/parser.py

```diff
@@ -184,52 +110,6 @@ def string_contents(self):
         return ast.literal_eval(self.text)
 
 
-@dataclass(frozen=True)
```

Comment:
also cut/pasted into span.py The two classes reference each other, and I didn't want to change the definitions, so I put them in the same file. Maybe the file should be called input.py. Maybe span_input.py. Happy to discuss :)